lib/Sema/SemaType.cpp

```diff
@@ -1565,6 +1565,15 @@ static QualType ConvertDeclSpecToType(TypeProcessingState &state) {
       }
       break;
   }
+  case DeclSpec::TST_exists: {
+    auto InnerType = S.GetTypeFromParser(DS.getRepAsType());
+    assert(DS.typeVariables().size() == 1 && "Expected exactly one type variable for an existential");
```

Comment:
Is this guaranteed?  I don't believe it is.  If it isn't, you shouldn't assert this, as it could cause a compiler crash.  Instead, there should be a diagnostic that this isn't supported yet.  You can use setInvalidType to invalidate the type.

Reply:
Yes, that there is just one type variable is enforced by the parser.